core/shared/src/main/scala/zio/Scope.scala

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+import scala.collection.immutable.LongMap
+
+trait Scope extends Serializable { self =>
+  def addFinalizer(finalizer: Exit[Any, Any] => UIO[Any]): UIO[Unit]
+  def close(exit: Exit[Any, Any]): UIO[Unit]
+  final def use[R, E, A](zio: ZIO[Scope with R, E, A]): ZIO[R, E, A] =
+    zio.provideSomeEnvironment[R](_.union[Scope](ZEnvironment(self))).onExit(self.close(_))
+}
+
+object Scope {
+
+  type Finalizer = Exit[Any, Any] => UIO[Any]
+
+  def addFinalizer(finalizer: Exit[Any, Any] => UIO[Any]): ZIO[Scope, Nothing, Unit] =
```

Comment:
Was just browsing through the code and noticed you could use the `Finalizer` alias here

Reply:
Yes. I'm not sure the type alias makes sense. It definitely makes this could shorter but it also adds an additional layer of indirection so may make sense to only use internally or to delete entirely.